kernel/src/process.rs

```diff
@@ -352,6 +352,29 @@ impl ProcessRestartPolicy for ThresholdRestart {
     }
 }
 
+/// Implementation of `ProcessRestartPolicy` that uses a threshold to decide
+/// whether to restart an app. If the app has been restarted more times than the
+/// threshold then the system will panic.
+pub struct ThresholdRestartThenPanic {
+    threshold: usize,
```

Comment:
Why is threshold a `usize`? It's just a positive number, so wouldn't `u32` (or honestly `u8`) fit better?

Reply:
I think because the stored reset count is a usize https://github.com/tock/tock/blob/5e39dc2bc69b5a07a2583f4e9d2629a9b04eed71/kernel/src/process.rs#L629

Reply:
Which was presumably an arbitrary choice back in #941

Reply:
`u8` and `u32` seem a bit _more_ arbitrary than `usize`... `usize` is saying that the count fits in a register. `u8` and `u32` says there is a particular maximum upper bound, but how do we chose which one?

Reply:
Well, in my mind `usize` refers to something that is architecture dependent and should change size with the architecture. Whereas if you have something that is just a number and has not relation to the architecture, you should be using a non-architecture-dependent type for it (`u8`, `u16`, `u32`) and making a decision about upper bounds.

Reply:
@brghena fair, that convinces me... merge for now unless @bradjc wants to change it.

Reply:
Yeah, for now at least restart counting is internally consistent.

Reply:
Haha nope :)